### PR TITLE
fix: add missing type defs to EDR NPM package

### PR DESCRIPTION
**Pull request by agostbiro**

### CODE REVIEW

File: .github/workflows/edr-npm-release.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ permissions:
   push:
     branches:
       - main
+      - edr/ci/fix-missing-type-def-in-edr-package
```
Comment (Wodann) [inline .github/workflows/edr-npm-release.yml:13]:
```suggestion
      - edr/main
```

Comment (agostbiro) [reply .github/workflows/edr-npm-release.yml]: Fixed in [bf4e7f2](https://github.com/NomicFoundation/hardhat/pull/4512/commits/bf4e7f20c6691262160e41e78ab064072ed753c8)

**Review by Wodann — APPROVED** [review]
LGTM, but there are two things to consider before merging:

- Could you please also upgrade the hardhat-core/package.json  to reflect the EDR version?
- We will likely want to be able to publish from `edr/main`

**Comment by agostbiro** [comment]
We decided to keep the carret version in Hardhat core.

**Pull request merged**